packages/mcfunction/src/parser/macro.ts

```diff
@@ -0,0 +1,153 @@
+import * as core from '@spyglassmc/core'
+import { localeQuote, localize } from '@spyglassmc/locales'
+import type { MacroChildNode, MacroKeyNode, MacroNode } from '../node/index.js'
+
+/**
+ * Parse a macro.
+ */
+export function macro(): core.Parser<MacroNode> {
+	return (
+		src: core.Source,
+		ctx: core.ParserContext,
+	): core.Result<MacroNode> => {
+		const start = src.cursor
+		const children: MacroChildNode[] = []
+
+		// Handle the starting '$'
+		if (src.peek() == '$' && src.peek(2) !== '$(') {
```

Comment:
Why do you need to check `src.peek() == '$'`? This parse function is only called when this condition is already true. And what is the purpose of checking for `$(` in the beginning of a macro line? It would be an invalid command but the parser shouldn't care about that.